src/pojo-interfaces.test.ts

```diff
@@ -0,0 +1,332 @@
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { describe, it, expect } from "@jest/globals";
+import { dataset } from "@rdfjs/dataset";
+import * as fc from "fast-check";
+import { DataFactory } from "n3";
+import {
+  serializeBoolean,
+  serializeDatetime,
+  serializeDecimal,
+  serializeInteger,
+  xmlSchemaTypes,
+} from "./datatypes";
+import { fromRdfJsDataset, toRdfJsDataset } from "./pojo-interfaces";
+
+describe("fromRdfJsDataset", () => {
+  const fcNamedNode = fc.webUrl().map((url) => DataFactory.namedNode(url));
+  const fcString = fc.string().map((value) => DataFactory.literal(value));
+  const fcInteger = fc
+    .integer()
+    .map((value) =>
+      DataFactory.literal(
+        serializeInteger(value),
+        DataFactory.namedNode(xmlSchemaTypes.integer)
+      )
+    );
+  const fcDecimal = fc
+    .float()
+    .map((value) =>
+      DataFactory.literal(
+        serializeDecimal(value),
+        DataFactory.namedNode(xmlSchemaTypes.decimal)
+      )
+    );
+  const fcDatetime = fc
+    .date()
+    .map((value) =>
+      DataFactory.literal(
+        serializeDatetime(value),
+        DataFactory.namedNode(xmlSchemaTypes.dateTime)
+      )
+    );
+  const fcBoolean = fc
+    .boolean()
+    .map((value) =>
+      DataFactory.literal(
+        serializeBoolean(value),
+        DataFactory.namedNode(xmlSchemaTypes.boolean)
+      )
+    );
+  const fcLangString = fc
+    .tuple(
+      fc.string(),
+      fc.oneof(fc.constant("nl-NL"), fc.constant("en-GB"), fc.constant("fr"))
+    )
+    .map(([value, lang]) => DataFactory.literal(value, lang));
+  const fcArbitraryLiteral = fc
+    .tuple(fc.string(), fc.webUrl())
+    .map(([value, dataType]) =>
+      DataFactory.literal(value, DataFactory.namedNode(dataType))
+    );
+  const fcLiteral = fc.oneof(
+    fcString,
+    fcInteger,
+    fcDecimal,
+    fcDatetime,
+    fcBoolean,
+    fcLangString,
+    fcArbitraryLiteral
+  );
+  // To check: not sure whether this generates new blank nodes every time.
+  const fcBlankNode = fc.constant(null).map(() => DataFactory.blankNode());
+  const fcDefaultGraph = fc.constant(DataFactory.defaultGraph());
+  const fcGraph = fc.oneof(fcDefaultGraph, fcNamedNode);
+  const fcQuadSubject = fc.oneof(fcNamedNode, fcBlankNode);
+  const fcQuadPredicate = fcNamedNode;
+  const fcQuadObject = fc.oneof(fcNamedNode, fcLiteral, fcBlankNode);
+  const fcQuad = fc
+    .tuple(fcQuadSubject, fcQuadPredicate, fcQuadObject, fcGraph)
+    .map(([subject, predicate, object, graph]) =>
+      DataFactory.quad(subject, predicate, object, graph)
+    );
+  const fcTerm = fc.oneof(fcNamedNode, fcLiteral, fcBlankNode, fcQuad);
+  const fcDataset = fc.set(fcQuad).map((quads) => dataset(quads));
+
+  it("can represent all Quads", () => {
+    const blankNode1 = DataFactory.blankNode();
+    const blankNode2 = DataFactory.blankNode();
+    const subject1Iri = "https://some.pod/resource#subject1";
```

Comment:
Since `subject1` must be an IRI (by virtue of being an RDF subject at all), I'd suggest naming this local var `subject1Str`, or `subject1AsString`:
```suggestion
    const subject1AsString = "https://some.pod/resource#subject1";
```

Reply:
e0cb294d49cf5a3f88f4ac8c0653d7ad057a04c9

Reply:
I added a comment on that commit (i.e., my suggestion was less work than the changes there :) !): https://github.com/inrupt/solid-client-js/commit/e0cb294d49cf5a3f88f4ac8c0653d7ad057a04c9#r48915352

Reply:
Responded here: https://github.com/inrupt/solid-client-js/pull/947#commitcomment-48932899

The change: 002ec66